BUG: DatetimeIndex.is_year_start breaks on custom business days frequencies bigger then `1C`

natmokval opened the pull request:
- [x] closes #58664
- [x] Tests added and passed
- [x] Added an entry in the latest `doc/source/whatsnew/v3.0.0.rst

natmokval committed:
fix bug dti.is_year_start breaks on freq custom business day with digit

natmokval committed:
Merge branch 'main' into bug-DatetimeIndex-is_year_start-breaks-on-freq-custom-business-day-with-digit

natmokval committed:
rename the variable freqstr

mroeschke commented on doc/source/whatsnew/v3.0.0.rst:
```diff
@@ -419,6 +419,7 @@ Interval
 Indexing
 ^^^^^^^^
 - Bug in :meth:`DataFrame.__getitem__` returning modified columns when called with ``slice`` in Python 3.12 (:issue:`57500`)
+- Bug in :meth:`DatetimeIndex.is_year_start` and :meth:`DatetimeIndex.is_quarter_start` does not raise on Custom business days frequencies bigger then "1C" (:issue:`58664`)
```
Could you put this under `Datetimelike`?

natmokval replied:
thanks, updated

mroeschke commented on pandas/_libs/tslibs/fields.pyx:
```diff
@@ -243,18 +243,17 @@ def get_start_end_field(
 
-        if (freqstr[0:2] in ["MS", "QS", "YS"]) or (
-                freqstr[1:3] in ["MS", "QS", "YS"]):
+        if freq_name.lstrip("B")[0:2]  in ["MS", "QS", "YS"]:
```
```suggestion
        if freq_name.lstrip("B")[0:2] in ["MS", "QS", "YS"]:
```

natmokval replied:
done

MarcoGorelli commented:
this needs rebasing, as it contains commits from the other PR (and could you move the whatsnew from that one to `DatetimeLike` please?)

natmokval committed:
resolve conflicts

natmokval committed:
delete double space

natmokval commented:
> this needs rebasing, as it contains commits from the other PR (and could you move the whatsnew from that one to `DatetimeLike` please?)
> 
thanks, rebased and moved the note in whatsnew from `Indexing` to `DatetimeLike`

mroeschke commented:
Thanks @natmokval
